### ISSUE TITLE
Comments on "Correct setup of OpenXR in a HoloLens 2 project including Holographic Remoting"

### ISSUE THREAD

**Issue by LocalJoost** [LocalJoost/BlogComments#389]
Original article: https://localjoost.github.io/Correct-setup-of-OpenXR-in-a-HoloLens-2-project-including-Holographic-Remoting/

**Comment by medinags** [comment]
**Hand tracking and interactions not working**
First of all thank you for your posts guiding us on the Augmented Reality road.

I am a AR young developer in HoloLens, in fact I come from using unity 2019 and MRTK 2.5 and as you know using legacy. 
I cloned your repository, in the editor, everything works fine. When I compile and deploy in H2 everything Okay. But, and here comes the problem. 
I have no hand tracking, and I certainly can't interact with anything.
I have followed the official microsoft documentation, but I am still stuck and get the same issue.

I'm using Unity 2020.3.19f1 LTS Version

**XR Plugin Manager**
![image](https://user-images.githubusercontent.com/78471817/135808421-b43d2046-354a-4850-aee5-68917aa891c7.png)
**Open XR.**
![image](https://user-images.githubusercontent.com/78471817/135808552-288662d0-4d7e-4f60-bbc8-e6904c2d01b3.png)
**MRTK Camera providers**
![image](https://user-images.githubusercontent.com/78471817/135808643-0e44fef8-1313-4eb1-b5e8-c5d220380fc2.png)
**MRTK Inputs Providers**
![image](https://user-images.githubusercontent.com/78471817/135808704-6509dae5-e492-4c6a-809f-a208146a8feb.png)
**My Packages**
![image](https://user-images.githubusercontent.com/78471817/135808779-cbae370d-444c-4f3d-a359-e89263b32bf4.png)

I don't understand, it all looks right to me.
I just hope, that you can help me. 
Thank you very much.

**Comment by LocalJoost** [comment]
Well it looks good to me too. Can you please open the generated solution in Visual Studio, open the package.appxmanifest and see what's here?
![image](https://user-images.githubusercontent.com/4129183/135819816-2424c169-032d-43ce-aa36-992894a1b0f3.png)
Also if you download my sample application showing hand interaction using the OpenXR plugin, does that work for you?

https://github.com/LocalJoost/HandCollider/tree/blog1

**Comment by medinags** [comment]
Actually that is your project, I have even downloaded the unity version which you used in your project.
Unity 2020.3.6f1

![image](https://user-images.githubusercontent.com/78471817/135825948-f1ae560c-53de-4b8b-a2f4-c26eaa1a8558.png)

Is enable. 
Gaze Input.
Internet (client)
Microphone
Spatial Perception
WebCam

In this video you can see how to run your project on my H2 device.

https://youtu.be/YnYQnGsezFU

As you can see, I don't have hand tracking, but eye tracking seems to work.

**Comment by LocalJoost** [comment]
I see. It does not work this way. That particular sample show a different mode of interaction. You should be able to push the cube, like I show in [this article](https://localjoost.github.io/Manipulating-holograms-using-the-MRTK-Hand-Physics-service/). The air tap won't work at all. 
See [this video](https://youtu.be/Jb3wkhpxnUc)

**Comment by medinags** [comment]
It's true, the air tap doesn't work at all. 

In the unity editor everything works fine

I have tried with different versions of MRTK.
Different tutorials and documentations and still no, I'm still stuck.
I have no idea what I am doing wrong. Or if it's a unity bug.
Do you have a video or a post on how to install MRTK 2.7 in Unity using OpenXR for the HoloLens?

**Comment by LocalJoost** [comment]
I am sorry, by now it's not clear to me what you are trying to achieve

**Comment by medinags** [comment]
I am sorry if I did not make myself clear.
I want to use the physical hand system. But I don't know why at the moment of running on the device it doesn't work at all.
I don't know what I'm doing wrong. 

I thought it is the MRTK or OpenXR version.
But no, whenever I run an application on the HoloLens that built on Unity 2020 using MRTK and OpenXR, I don't get the hand tracking. 

Even the repository of your project.
I have tried it by remoting mode and in that case I get the hand tracking and I can interact with the cube.

**Comment by LocalJoost** [comment]
So to recap:

- You have cloned this repository/branch https://github.com/LocalJoost/HandCollider/tree/blog1
- If you run this code using Holographic Remoting you can interact with the cube, on the HoloLens
- If you build the app and deploy it to the HoloLens and run it, it does NOT work. You cannot interact with the cube. 

Correct?

**Comment by medinags** [comment]
Correct, Afeter build, deploy and run, I have no interaction. The hand tracking does not work at all. Neither does the air tap.

**Comment by LocalJoost** [comment]
So I cloned my repo from scratch, built and deployed it...

https://user-images.githubusercontent.com/4129183/135837932-83ccc933-456e-48ec-868a-048783136a55.mp4

**Comment by LocalJoost** [comment]
These are my settings in Visual Studio that I use to build an deploy:
![image](https://user-images.githubusercontent.com/4129183/135838932-b21cef38-8f37-4a0a-8c68-8cac44176bc9.png)
![image](https://user-images.githubusercontent.com/4129183/135839002-6e8b655f-7d78-4b34-8d48-9c53cfd2fee2.png)

**Comment by medinags** [comment]
Same issue. No hand tracking
Unity Build 
![image](https://user-images.githubusercontent.com/78471817/135840452-8db6557c-e4c7-4f38-b107-aa5887c47b26.png)
Unity Console
![image](https://user-images.githubusercontent.com/78471817/135840765-39ab11b9-ece7-43ba-8037-e2b62b1bfb4a.png)
VS
![image](https://user-images.githubusercontent.com/78471817/135840829-5eb0bca6-f84b-4eaf-a7d1-c899faf6bb1e.png)
![image](https://user-images.githubusercontent.com/78471817/135840855-2b7f3ac8-ed4b-4446-a314-4487026b71aa.png)
![image](https://user-images.githubusercontent.com/78471817/135841176-1349eb1f-ede6-4177-8466-d387972baf11.png)

**Comment by medinags** [comment]
Final result.
This happens when I launch the application.

https://youtu.be/7SWRyM5RadY

As you can see, I have no hand tracking at all.
Something strange is that it seems as if it is launched twice.

**Comment by LocalJoost** [comment]
Okay. My last attempt. Uninstall the app as it is now from the HoloLens.
Download the package I created [here ](https://1drv.ms/u/s!AjesCJDYqwYk2ZxNKPabfIW3Gy0n_A?e=2EqlgC) and install it using the device portal.
Start making a movie before you start it the very first time. I would like to see what happens and how you launch it the first time.

**Comment by medinags** [comment]
**Et voilà !**

When I was in the device portal. I noticed that there was a field OpenXR.
Which requested an update. And as I thought about it. 
I updated everything. 

![image](https://user-images.githubusercontent.com/78471817/135843720-288d339b-19c5-4aae-b6e1-e8c4885c69d7.png)

And then I deployed and ran my solution. 
And everything works perfectly.

**So the solution was the Open XR update on divice portal.** 
Again thank you very much for your time.
And above all for accompanying people who are just starting on this exciting journey.

**Comment by LocalJoost** [comment]
LOL well glad that - in a very roundabout way I more or less helped you stumble toward the solution. Happy developing!

**Comment by JCMazza** [comment]
Hi @LocalJoost, 

Aplogies I asked some question related to this post in the wrong place which ended up confusing you and I was also not very clear.

Basically the manual settings we set in the XR plugin manager for hololens on UWP an standalone I want to be able to do that via an editor script in code. I am able to set via the openxr api the different interaction profiles and even the meta features for android but I can not figure out how to set the "Microsoft Hololens" feature option.

Example for android I do something like this

switch (_targetDeviceOptions[_targetDeviceIndex])
            {
       
                case "OculusQuest":
                    XRPackageMetadataStore.AssignLoader(androidSettings.Manager, OPENXR_LOADER, BuildTargetGroup.Android);
                    XRPackageMetadataStore.AssignLoader(standaloneSettings.Manager, OPENXR_LOADER, BuildTargetGroup.Standalone);
                    SetOpenXRSettingsForMetaQuest(true);
                    SwitchToAndroid();
                    break;
            }
            
private void SetOpenXRSettingsForMetaQuest(bool enable)
        {
            var metaQuestSupportFeatureAndroid = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android).GetFeature(typeof(MetaQuestFeature));
            metaQuestSupportFeatureAndroid.enabled = enable;
            
            var metaQuestInteractionProfileAndroid = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android).GetFeature(typeof(OculusTouchControllerProfile));
            metaQuestInteractionProfileAndroid.enabled = enable;
            
            var metaQuestInteractionProfileStandalone = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Standalone).GetFeature(typeof(OculusTouchControllerProfile));
            metaQuestInteractionProfileStandalone.enabled = enable;
        }

As you can see OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android).GetFeature(typeof(MetaQuestFeature)); would tick the Meta quest feature under OpenXR -> Android but I can not find any "typeof" that exist for the hololens (standalone and WSA)

**Comment by LocalJoost** [comment]
Ah, I see. TBH, I have no idea. I usually go for a very low tech solution. I do the Unity project settings so it builds for HoloLens. Then I commit the whole shebang to Git. Then I make the app build for Quest, and see what files are changed. I copy the changed files to a separate folder and commit them to Git, in that separate folder. A simple pull makes it compilable for HoloLens, if pull and then copy over the files from the Quest folder, it builds for Quest.
